### PR TITLE
Bluetooth: Mesh: config options to use tfm and oberon

### PR DESCRIPTION
**Pull request by alxelax**
PR adds configurtion options to use tf-m and Oberon as a default crypto engine for non-secure platforms
in mesh samples.

### CODE REVIEW

File: subsys/bluetooth/mesh/Kconfig [hunk 1]
```diff
@@ -109,6 +109,38 @@ config BT_MESH_PROXY_FILTER_SIZE
 
 endif
 
+if BT_MESH_USES_TFM_PSA
```
Comment (joerchan) [inline subsys/bluetooth/mesh/Kconfig:112]: config BT_MESH_USES_TFM_PSA
     bool
     select PSA_WANT_dep1
     select PSA_WANT_dep2

File: subsys/bluetooth/mesh/Kconfig [hunk 2]
```diff
@@ -109,6 +109,38 @@ config BT_MESH_PROXY_FILTER_SIZE
...
+
+choice TFM_PROFILE_TYPE
+	default TFM_PROFILE_TYPE_NOT_SET
+endchoice
+
+config PSA_CRYPTO_DRIVER_OBERON
+	default y
```
Comment (joerchan) [inline subsys/bluetooth/mesh/Kconfig:119]: Please remove this, it's default y already.
We will make crypto_driver selection automatic based on PSA_WANT symbols, so the protocol shouldn't have to worry about this.

**Review by joerchan — CHANGES_REQUESTED** [review]

**Review by joerchan — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Comment by alxelax** [comment]
> Seems like thingy53_nrf5340_cpuapp_ns light_ctrl sample ran out of flash by quite a great deal according to CI. Do you know why?

Not yet. I try to figure out.

**Review by PavelVPV — APPROVED** [review]

**Comment by joerchan** [comment]
> Seems like thingy53_nrf5340_cpuapp_ns light_ctrl sample ran out of flash by quite a great deal according to CI. Do you know why?

Thingy53 uses static partition layout and it's size is set to match TFM MINIMAL profile type.
So either this needs to set a new static partition file to support the crypto config of BT-Mesh, or the thingy53 NS should not be supported in these samples.

**Comment by PavelVPV** [comment]
I guess changelog is required now

**Comment by alxelax** [comment]
> I guess changelog is required now

What do you mean? just label or anything else?

**Comment by PavelVPV** [comment]
> > I guess changelog is required now
> 
> What do you mean? just label or anything else?

https://github.com/nrfconnect/sdk-nrf/blob/2cfd6275046ba090460b19356563198084902cad/doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst?plain=1#L230-L235

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -89,7 +89,7 @@ Bluetooth® LE
 Bluetooth mesh
 --------------
 
-|no_changes_yet_note|
+* Added support of Trusted Firmware-M(TF-M) PSA as crypot backend for mesh security toolbox for ``_ns`` platforms.
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:92]:
```suggestion
* Added support for Trusted Firmware-M (TF-M) PSA as the crypto backend for mesh security toolbox for the platforms with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>`.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -226,6 +226,26 @@ Bluetooth mesh samples
 
   * Fixed an issue with the sample not fitting into RAM size on the nrf52dk_nrf52832 board.
 
+* :ref:`bluetooth_mesh_light` sample:
+
+  * Removed support of non-secure configuration for :ref:`zephyr:thingy53_nrf5340`.
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:231]:
```suggestion
  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.
```
And similarly for other entries.

**Review by greg-fer — CHANGES_REQUESTED** [review]
Some time ago we reworked how we talk about _ns build targets, also to make it clear that _non-secure_ is actually more secure than _secure_ :) This is the reason for edits, which now also point to the section that explains the difference.

**Review by PavelVPV — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**